tests/nlu/featurizers/test_lexical_syntactic_featurizer.py

```diff
@@ -195,3 +195,93 @@ def test_text_featurizer_using_pos_with_action_text(
 
     assert seq_vec is None
     assert sen_vec is None
+
+
+@pytest.mark.parametrize(
+    "sentence, featurizer_config",
+    [
+        (
+            "At 0700 the SUN is shining",
+            [
+                ["prefix2", "suffix1", "suffix3", "pos", "upper"],
+                [
+                    "low",
+                    "title",
+                    "prefix2",
+                    "suffix1",
+                    "suffix3",
+                    "pos",
+                    "upper",
+                    "digit",
+                ],
+                ["prefix2", "suffix1", "suffix3", "pos", "upper"],
+            ],
+        )
+    ],
+)
+def test_text_featurizer_using_feature_to_idx_dict(
```

Comment:
Can we break this test into multiple tests since it's testing too many things at once IMO?
If we want to test the memoization implementation I suggest we write a unit test just for that. To do that, it would be better if we extract out the memoization code lines 192-212 in `_tokens_to_features` to a separate function so that we can test it separately.

Reply:
@dakshvar22 There aren't any direct tests of the memoization: if memoization is implemented correctly, then the behavior of the feature is unchanged, but it's faster. The existing tests check behavior; I verified the speedup with cProfile. 

So technically the test is unrelated to the change. However, it exercises more parts of the code than the other tests and doesn't rely on any expected outputs, as in the other tests. By specifying those outputs, the other tests create undocumented (afaict) requirements on the order of the features, overconstraining the implementation. This test does not have any expected output. Rather, it checks whether the features are internally consistent, irrespective of how the features are ordered.

Reply:
I guess then the question is whether this test should be separated out or removed altogether?